rpc-client/src/nonblocking/rpc_client.rs

```diff
@@ -3535,6 +3535,49 @@ impl RpcClient {
             .await
     }
 
+    /// Returns a list of minimum prioritization fees from recent blocks.
+    /// Takes an optional vector of addresses; if any addresses are provided, the response will
+    /// reflect the minimum prioritization fee to land a transaction locking all of the provided
+    /// accounts as writable.
+    ///
+    /// Currently, a node's prioritization-fee cache stores data from up to 150 blocks.
+    ///
+    /// # RPC Reference
+    ///
+    /// This method corresponds directly to the [`getRecentPrioritizationFees`] RPC method.
+    ///
+    /// [`getRecentPrioritizationFees`]: https://docs.solana.com/developing/clients/jsonrpc-api#getrecentprioritizationfees
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use solana_rpc_client_api::client_error::Error;
+    /// # use solana_rpc_client::nonblocking::rpc_client::RpcClient;
+    /// # use solana_sdk::signature::{Keypair, Signer};
+    /// # futures::executor::block_on(async {
+    /// #     let rpc_client = RpcClient::new_mock("succeeds".to_string());
+    /// #     let alice = Keypair::new();
+    /// #     let bob = Keypair::new();
+    /// let addresses = vec![alice.pubkey(), bob.pubkey()];
+    /// let prioritization_fees = rpc_client.get_recent_prioritization_fees(
+    ///     &addresses,
+    /// ).await?;
+    /// #     Ok::<(), Error>(())
+    /// # })?;
+    /// # Ok::<(), Error>(())
+    /// ```
+    pub async fn get_recent_prioritization_fees(
+        &self,
+        addresses: &[Pubkey],
```

Comment:
wdyt about also adding a `get_recent_prioritization_fees_for_message()`? This would extract the writable accounts for that `Message` and then call `get_recent_prioritization_fees()` with them.

Reply:
a convenient helper to get the min, mean, max fee from a `Vec<RpcPrioritizationFee>` seems nice too?

I was basically thinking that a priority-aware client would just want to pass in a `Message` and use the `mean` or `max` returned fee, given it doesn't exceed some sanity threshold amount that it defines

Reply:
Yes, I like the idea of `RpcClient::get_recent_prioritization_fees_for_message()`.
Not sure the best place for a helper like you describe, though. Do you have any ideas?

I'm not sure if you're suggesting a single one-stop-shop client api that takes a message, and just returns max and mean fees. Are you? We tend to do that sort of thing in solana-cli and keep RpcClient apis simpler.

How do you feel about me merging these basic wrapper apis, and adding the more fully fledged fns in a second PR?

Reply:
> How do you feel about me merging these basic wrapper apis, and adding the more fully fledged fns in a second PR?

Yep, of course.

I was just trying to get ahead of what I expect clients to be doing. I guess they'd actually likely have a slice of `Instruction`s instead of a `Message`, since once the priority fee is figured they need to add the additional compute budget instruction to use it before constructing a `Message`. Yeah, an internal `solana-cli` helper if/when we add priority fees there makes sense first, and then once it feels good then deciding if it should be exposed as a public utility somehow

Reply:
> I guess they'd actually likely have a slice of `Instruction`s instead of a `Message`

Makes sense. From the perspective of solana-cli, the Message would be nicer, since we already have tooling to decode and deserialize Messages. RpcClient could, of course, support either or both.

Reply:
> RpcClient could, of course, support either or both.

I like both (ie. two different client methods), the more I think about it.